### DISCUSSION TITLE
CBLLE compilation doesn't work

### DISCUSSION THREAD

**Discussion by bourgeoisp** [codefori/vscode-ibmi#2378] [Q&A] [answered]
When I try to compile a CBLLE source (VERIFY.CBLLE in QGPL/QCBLLESRC), after modifying the following line:
![image](https://github.com/user-attachments/assets/50c33619-435d-4ee5-b933-ce20768acf67)

1. On the **TERMINAL** view and the OUTPUT file, all is OK, I got a severity 30 error:
![image](https://github.com/user-attachments/assets/79616eeb-8328-4bfc-8f86-ad40fdb6aba1)

2. But on on the **PROBLEMS** view, this severity 30 **error doesn't appear** (I don't use filtering) and some information errors appear:
![image](https://github.com/user-attachments/assets/2caa1922-0f97-4fa3-8fc3-26b819925344)

3. If I click on one of this information error, the following screen is displayed:
![image](https://github.com/user-attachments/assets/10277da0-696b-44e8-a68a-e475c0517ca2)

4. And if I click on the **"Clear Diagnostics"** button, the PROBLEMS view **is not cleared**...

**Comment by worksofliam** [comment]
I was able to recreate. Looks like something unique with COBOL. I'm finding out!

<img width="1412" alt="image" src="https://github.com/user-attachments/assets/2e3b78d8-1d09-4dbc-8512-8aa1222a4063">

**Comment by worksofliam** [comment]
@bourgeoisp it was really easy. I had to add `*EVENTF` to `OPTION` in my Action. It's not in there by default!

<img width="1412" alt="Screenshot 2024-11-27 at 8 18 00 AM" src="https://github.com/user-attachments/assets/5b720ede-7f6f-4043-9b64-7a8d2dd26865">

**Comment by bourgeoisp** [comment, marked as answer]
Yes, sure, thank you Liam... you're right, it wasn't complicated... sorry, I should have thought of that, it's the same as in RDi...

**Comment by bourgeoisp** [comment]
But it doesn't seem to work with CBL source even with the OPTION(*SRCDBG) parameter in the CRTCBLPGM command.

**Reply by worksofliam** [reply]
I don't believe CRTCBLPGM ever supported `*EVENTF` and there's not much we can do about that. :(

**Reply by bourgeoisp** [reply]
Yes I know, but it works in RDi:

![image](https://github.com/user-attachments/assets/650a8db4-4eb8-4ee8-ac5e-d8e334b57e24)

**Reply by bourgeoisp** [reply]
But don't bother investigating this case because it's easy to change a CBL source to a CBLLE source
